### PR TITLE
Add user-attributes

### PR DESCRIPTION
**Pull request by nsklikas**
@NicolasLiampotis

### CODE REVIEW

File: website/docs/user_attributes.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+---
+id: user_attributes
+title: User Attributes
+---
+
+## User attributes
+
+This section defines the attributes that can be made available to services
+connected to Check-in.
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: connected to the GRNET eID Proxy

File: website/docs/user_attributes.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+<!-- markdownlint-disable line-length no-inline-html -->
+
+|          attribute name | Unique ID                                                                                       |
+| ----------------------: | :---------------------------------------------------------------------------------------------- |
+|         **description** | An unique identifier for the user                       |
+|   **SAML Attribute(s)** | `urn:oid:1.3.6.1.4.1.5923.1.1.1.13` (eduPersonUniqueId)                                         |
+|          **OIDC scope** | `openid`                                                                                        |
+|       **OIDC claim(s)** | `sub`                                                                                           |
+| **OIDC claim location** | <ul><li>ID token</li><li>Userinfo endpoint</li><li>Introspection endpoint</li></ul>             |
+|              **origin** | Check-in assigns this attribute on user registration                                            |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: origin: Assigned by the GRNET eID Proxy

File: website/docs/user_attributes.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+|             **changes** | No                                                                                              |
+|        **multiplicity** | No                                                                                              |
+|        **availability** | Always                                                                                          |
+|             **example** | _4a58cb0338fb68060ba75e20e02a4844148f9c63a5cd9a7418dc5a9a6ab641cbaef73b4c65519286359f766c45bdbad36cc9189e75a03bd12f13526c34cab93e@eid-proxy.aai.grnet.gr_   |
+|               **notes** | Use **EGI ID** within your application as the unique-identifier key for the user                |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: `Use this attribute within your application as the unique-identifier key for the user`

File: website/docs/user_attributes.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+This section defines the attributes that can be made available to services
+connected to Check-in.
+
+### 1. Unique ID
+
+<!-- markdownlint-disable line-length no-inline-html -->
+
+|          attribute name | Unique ID                                                                                       |
+| ----------------------: | :---------------------------------------------------------------------------------------------- |
+|         **description** | An unique identifier for the user                       |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: A unique identifier for the user

File: website/docs/user_attributes.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/PersonIdentifier` (PersonIdentifier) |
+|          **OIDC scope** | `profile`                                                                           |
+|       **OIDC claim(s)** | `person_identifier`                                                                 |
+| **OIDC claim location** | Userinfo endpoint                                                                   |
+|              **origin** | Provided by user's Identity Provider                                                |
+|             **changes** | Yes                                                                                 |
+|        **multiplicity** | Single-valued                                                                       |
+|        **availability** | Always                                                                              |
+|             **example** | _CA/CA/UniqueId1_                                                                   |
+|               **notes** | -                                                                                   |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: notes: Any service that consumes assertions of identity must assume that the Unique Identifier presented for a particular person may change over time e.g. where the user's digital identity is replaced or repaired. This should be handled by a consuming service using the same matching process as used when an identity is first encountered utilising the Minimum Data Set to identify the Principal within the service.

File: website/docs/user_attributes.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+### 2. Person Identifier
+
+|          attribute name | Person Identifier                                                                   |
+| ----------------------: | :---------------------------------------------------------------------------------- |
+|         **description** | The user's person identifier                                                        |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/PersonIdentifier` (PersonIdentifier) |
+|          **OIDC scope** | `profile`                                                                           |
+|       **OIDC claim(s)** | `person_identifier`                                                                 |
+| **OIDC claim location** | Userinfo endpoint                                                                   |
+|              **origin** | Provided by user's Identity Provider                                                |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: origin: Provided by the eIDAS Node from which the user is authenticating

File: website/docs/user_attributes.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+|             **changes** | Yes                                                                                 |
+|        **multiplicity** | Single-valued                                                                       |
+|        **availability** | Always                                                                              |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: availability: Always (for natural person authentication through eIDAS)

File: website/docs/user_attributes.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+|              **status** | Stable                                                                                          |
+
+<!-- markdownlint-enable line-length no-inline-html -->
+
+### 2. Person Identifier
+
+|          attribute name | Person Identifier                                                                   |
+| ----------------------: | :---------------------------------------------------------------------------------- |
+|         **description** | The user's person identifier                                                        |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/PersonIdentifier` (PersonIdentifier) |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: Question: Is this forwarded by the eID Proxy to SAML2int SPs as is or mapped to an eduPerson attribute?

File: website/docs/user_attributes.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+### 3. Given Name
+
+|          attribute name | Given Name                           |
+| ----------------------: | :----------------------------------- |
+|         **description** | The user's first name                |
+|   **SAML Attribute(s)** | `urn:oid:2.5.4.42` (givenName)       |
+|          **OIDC scope** | `profile`                            |
+|       **OIDC claim(s)** | `given_name`                         |
+| **OIDC claim location** | Userinfo endpoint                    |
+|              **origin** | Provided by user's Identity Provider |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md:57]: Question: Is this mapped from the eIDAS FirstName attribute in the case of natural person authentication through eIDAS?

File: website/docs/user_attributes.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+### 4. Family Name
+
+|          attribute name | Family Name                          |
+| ----------------------: | :----------------------------------- |
+|         **description** | The user's last name                 |
+|   **SAML Attribute(s)** | `urn:oid:2.5.4.4` (sn)               |
+|          **OIDC scope** | `profile`                            |
+|       **OIDC claim(s)** | `family_name`                        |
+| **OIDC claim location** | Userinfo endpoint                    |
+|              **origin** | Provided by user's Identity Provider |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md:74]: Question: Is this mapped from the eIDAS FamilyName attribute in the case of natural person authentication through eIDAS?

File: website/docs/user_attributes.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+### 5. Birth Date
+
+|          attribute name | Birth Date                                           |
+| ----------------------: | :--------------------------------------------------- |
+|         **description** | The user's birth date                                |
+|   **SAML Attribute(s)** | `urn:oid:1.3.6.1.4.1.25178.1.2.3` (schacDateOfBirth) |
+|          **OIDC scope** | `profile`                                            |
+|       **OIDC claim(s)** | `birth_date`                                         |
+| **OIDC claim location** | Userinfo endpoint                                    |
+|              **origin** | Provided by user's Identity Provider                 |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md:91]: Question: Is this mapped from the eIDAS `DateOfBirth` attribute in the case of natural person authentication through eIDAS?

File: website/docs/user_attributes.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+|         **description** | The user's legal address                                                                        |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/LegalAddress` (LegalAddress)                     |
+|          **OIDC scope** | `legal_address`                                                                                 |
+|       **OIDC claim(s)** | `legal_address`                                                                                 |
+| **OIDC claim location** | Userinfo endpoint                                                                               |
+|              **origin** | Provided by user's Identity Provider                                                            |
+|             **changes** | Yes                                                                                             |
+|        **multiplicity** | Single-valued                                                                                   |
+|        **availability** | Always                                                                                          |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: availability: Only when released by the eIDAS Node from which the legal person is authenticating

File: website/docs/user_attributes.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+|        **availability** | Always                                                                                        |
+|             **example** | _CA/CA/UniqueId1_                                                                             |
+|               **notes** | -                                                                                             |
+|              **status** | Stable                                                                                        |
+
+### 9. Legal Address
+
+|          attribute name | Legal Name                                                                                      |
+| ----------------------: | :---------------------------------------------------------------------------------------------- |
+|         **description** | The user's legal address                                                                        |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: description: The current address for the natural person as registered with the eIDAS Member State authority.

File: website/docs/user_attributes.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+|          **OIDC scope** | `legal_address`                                                                                 |
+|       **OIDC claim(s)** | `legal_address`                                                                                 |
+| **OIDC claim location** | Userinfo endpoint                                                                               |
+|              **origin** | Provided by user's Identity Provider                                                            |
+|             **changes** | Yes                                                                                             |
+|        **multiplicity** | Single-valued                                                                                   |
+|        **availability** | Always                                                                                          |
+|             **example** | _PGVpZGFzLWxlZ2FsOkZ1bGxDdmFkZHJlc3M+TGVnYWwgYWRkcmVzczwvZWlkYXMtbGVnYWw6RnVsbEN2YWRkcmVzcz4K_  |
+|               **notes** | -                                                                                               |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: notes: Address data is structured as an XML sequence of xsd:string elements. This data is base64 encoded.

File: website/docs/user_attributes.md [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+|         **description** | The user's vat registration number                                                             |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/VATRegistrationNumber` (VATRegistrationNumber)  |
+|          **OIDC scope** | `vat_registration`                                                                             |
+|       **OIDC claim(s)** | `vat_registration`                                                                             |
+| **OIDC claim location** | Userinfo endpoint                                                                              |
+|              **origin** | Provided by user's Identity Provider                                                           |
+|             **changes** | Yes                                                                                            |
+|        **multiplicity** | Single-valued                                                                                  |
+|        **availability** | Always                                                                                         |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: availability: Only when released by the eIDAS Node from which the legal person is authenticating

File: website/docs/user_attributes.md [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+|             **example** | _PGVpZGFzLWxlZ2FsOkZ1bGxDdmFkZHJlc3M+TGVnYWwgYWRkcmVzczwvZWlkYXMtbGVnYWw6RnVsbEN2YWRkcmVzcz4K_  |
+|               **notes** | -                                                                                               |
+|              **status** | Stable                                                                                          |
+
+### 10. Vat Registration
+
+|          attribute name | Vat Registration                                                                               |
+| ----------------------: | :--------------------------------------------------------------------------------------------- |
+|         **description** | The user's vat registration number                                                             |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/VATRegistrationNumber` (VATRegistrationNumber)  |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: Question: Is this forwarded by the eID Proxy to SAML2int SPs as is or mapped to some other attribute?

File: website/docs/user_attributes.md [hunk 17]
```diff
@@ -0,0 +1,166 @@
...
+|         **description** | The user's legal person identifier                                                            |
+|   **SAML Attribute(s)** | `http://eidas.europa.eu/attributes/legalperson/LegalPersonIdentifier` (LegalPersonIdentifier) |
+|          **OIDC scope** | `legal_profile`                                                                               |
+|       **OIDC claim(s)** | `legal_person_identifier`                                                                     |
+| **OIDC claim location** | Userinfo endpoint                                                                             |
+|              **origin** | Provided by user's Identity Provider                                                          |
+|             **changes** | Yes                                                                                           |
+|        **multiplicity** | Single-valued                                                                                 |
+|        **availability** | Always                                                                                        |
```
Comment (NicolasLiampotis) [inline website/docs/user_attributes.md]: availability: Always (for legal person authentication through eIDAS)

**Review by NicolasLiampotis — CHANGES_REQUESTED** [review]
Added suggestions and questions

File: website/docs/user_attributes.md (hunk 8 above)
Comment (nsklikas) [reply website/docs/user_attributes.md]: It is forwarded as is

Comment (nsklikas) [reply website/docs/user_attributes.md]: We currently have no SAML SPs connected that get legal person attributes so it is null for now

**Review by NicolasLiampotis — APPROVED** [review]
PLGTM

**Pull request merged**